### ISSUE TITLE
Apple silicon Macs and "Use the new Virtualization framework" experimental feature option

### ISSUE THREAD

**Issue by jamesdh** [docker/for-mac#6061]
TLDR: It is my understanding that M1 Macs don't even have the option for using the old virtualization framework, so don't even expose it as an option under Experimental Features. 

Longer version: 
Docker has previously indicated that Docker for Mac on M1-based Macs would ["move onto Apple’s new hypervisor framework"](https://www.docker.com/blog/apple-silicon-m1-chips-and-docker/). By this, I assume they originally meant the Big Sur virtualization framework, which would imply the old virtualization framework should not even be an option.

I raise this because I'm experiencing a bug with `docker exec -d` on both an M1 Mac as well as an Intel Mac that had the "Use new Virtualization framework" checked. When disabling the new virtualization framework on the Intel Mac, the bug goes away. Disabling this option on the M1 Mac does nothing, and the bug remains. My assumption is that despite how I configure it on my M1 Mac, it never actually changes and instead just uses the newer virtualization framework. This is confusing. 

Once I find a simple way to recreate the `docker exec -d` bug I'm experiencing, I'll create another issue and link it here as well.

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
If using the old virtualization framework truly is not an option, it should be presented as such.

### Actual behavior
Config option is still there but appears to do nothing. 

### Information
  - macOS Version: 12.0.1
  - Intel chip or Apple chip: Apple
  - Docker Desktop Version: 4.2.0

**Comment by jamesdh** [comment]
So what we thought was a bug w/ `docker exec -d` on the new virtualization framework ended up being a bug with one of our scripts that made a false assumption as to how logging worked on such detached sessions. It worked on the old framework on the Intel Mac I believe because of some increased latency in starting our image. But this still begs the question, is the underlying virtualization framework actually selectable on M1 Mac's?

**Comment by mat007** [comment]
That’s right, the «old» virtualization framework just doesn’t exist on Apple Silicon.

**Comment by grahl** [comment]
Without the checkbox set in 4.2.0 I'm seeing significantly different performance behavior (faster IO, slower execution) and it seems to use qemu-system-aarch64 underneath then.

**Comment by memoryleak** [comment]
> Without the checkbox set in 4.2.0 I'm seeing significantly different performance behavior (faster IO, slower execution) and it seems to use qemu-system-aarch64 underneath then.

I can confirm the performance issues with the experimental hypervisor feature:

- Using the experimental hypervisor, a 80 megabytes MySQL dump takes 1.5 hours to import
- Using the non-experimental hypervisor, it took 38 seconds.

**Comment by vembry** [comment]
> > Without the checkbox set in 4.2.0 I'm seeing significantly different performance behavior (faster IO, slower execution) and it seems to use qemu-system-aarch64 underneath then.
> 
> I can confirm the performance issues with the experimental hypervisor feature:
> 
> * Using the experimental hypervisor, a 80 megabytes MySQL dump takes 1.5 hours to import
> * Using the non-experimental hypervisor, it took 38 seconds.

i can also confirm this

**Comment by grEvenX** [comment]
We just recently discovered an issue like you are describing, and I wonder if it could be related to the to macOS 12.1 update.
@memoryleak @vembry Did you by any chance have macOS 12.1 beta installed at the time you experienced this issue?

**Comment by memoryleak** [comment]
Hi @grEvenX
No, I never run Beta version of Mac, however I can't remember if 12.1 was installed at that time already or if it was afterwards.

**Comment by vembry** [comment]
hi @grEvenX , i think mine is still 11.4
![image](https://user-images.githubusercontent.com/22610742/146628332-758d729d-a912-43c4-b297-18c56695b127.png)

**Comment by henrik242** [comment]
Any updates on this?  Does the `Use the new Virtualization framework`  checkbox worsen the performance in all cases (old os, new os, x86, m1)?

**Comment by grEvenX** [comment]
@henrik242 yes, there has been a lot of recent development related to this topic. Please follow the comments from this post for more information: https://github.com/docker/roadmap/issues/7#issuecomment-1042577684

**Comment by jamesdh** [comment]
To be clear, the original reason for creating this issue was simply due to the confusion around enabling/disabling  Virtualization.framework on Apple silicon Mac's. Performance issues regarding Virtualization.framework is another issue entirely, and there are numerous issues that have been created to address this.

**Comment by dstranathan** [comment]
To clarify - if Docker can/will use Apples virtualization framework, does thta mean that Docker would talk directly to the Apple Darwin/XNU kernel - and not require a Linux sub-layer?

**Comment by reuben** [comment]
On M1 the checkbox toggles between QEMU and Virtualization.framework, so it is doing something. On Intel Macs Docker uses the old Hypervisor.framework (via HyperKit).
